src/Symfony/Component/Notifier/Channel/EmailChannel.php

```diff
@@ -57,7 +57,7 @@ public function notify(Notification $notification, RecipientInterface $recipient
         if ($email instanceof Email) {
             if (!$email->getFrom()) {
                 if (null === $this->from) {
-                    throw new LogicException(sprintf('To send the "%s" notification by email, you should either configure a global "from" or set it in the "asEmailMessage()" method.', get_debug_type($notification)));
+                    throw new LogicException(sprintf('To send the "%s" notification by email, you should either configure a global envelope sender or set it in the "asEmailMessage()" method.', get_debug_type($notification)));
```

Comment:
Hmm before it was more clear to me, maybe we should not improve the message that way, but keen the `"from"` and add more information where one can do this?

cc @jschaedl

Reply:
The envelope sender is only a fallback:
https://github.com/symfony/symfony/blob/e7ed342ab9d3dfd33510e8f93f17929d3548ee7f/src/Symfony/Component/Notifier/Channel/EmailChannel.php#L44

What about:

```suggestion
                    throw new LogicException(sprintf('To send the "%s" notification by email, you should either configure a global "from", set a sender in the global "envelope" or set a "from" in the "asEmailMessage()" method.', get_debug_type($notification)));
```

Reply:
Yes its a fallback on L44 of this service but the default way the framework extension is setting the from argument in https://github.com/symfony/symfony/blob/5.x/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L2396.

What about this (a bit more verbose)?
```suggestion
                    throw new LogicException(sprintf('To send the "%s" notification by email, you should either configure a global "from", set a sender in the global "envelope" of the mailer configuration or set a "from" in the "asEmailMessage()" method.', get_debug_type($notification)));
```

Reply:
LGTM